### PR TITLE
GraphQL improvements for TSDoc and Playground/Schema Introspection using Envelop

### PR DESCRIPTION
**Pull request by dthyresson**
This is includes some minor improvements to the GraphQL servers - Apollo and Envelop.

* Adds tsdoc to the GraphQL handler and options. 
* Renames LoggerOptions to GraphQLLoggerOptions for clarity (and no confusion beter Logger options like log level, pretty printing, etc).

In Envelop:

* Properly disables *both* schema inspection & playground

The issue was that schema introspection was disable, but the rule Helix uses to check if the playground should launch was too lenient. Now the playground only launches in dev.

This behavior matches Apollo.

### CODE REVIEW

File: packages/api/src/functions/graphql.ts [hunk 1]
```diff
@@ -37,24 +37,67 @@ type DepthLimitOptions = { maxDepth: number; ignore?: string[] }
 /**
  * Options for request and response information to include in the log statements
  * output by UseRedwoodLogger around the execution event
+ *
+ * @param data - Include response data sent to client.
+ * @param operationName - Include operation name.
+ * @param requestId - Include the event's requestId, or if none, generate a uuid as an identifier.
+ * @param query - Include the query. This is the query or mutation (with fields) made in the request.
+ * @param tracing - Include the tracing and timing information.
+ * @param userAgent -Include the browser (or client's) user agent.
```
Comment (thedavidprice) [inline packages/api/src/functions/graphql.ts:46]:
```suggestion
 * @param userAgent - Include the browser (or client's) user agent.
```

File: packages/api/src/functions/graphql.ts [hunk 2]
```diff
@@ -224,54 +267,71 @@ export const createContextHandler = (
...
 /**
  * Creates an Apollo GraphQL Server.
  *
+ * @param options - GraphQLHandlerOptions
+ *
+ * @example
  * ```js
- * export const handler = createGraphQLHandler({ schema, context, getCurrentUser })
+ * export const handler = createGraphQLHandler({
+ *  loggerConfig: { logger, options: {} },
```
Comment (thedavidprice) [inline packages/api/src/functions/graphql.ts:325]: is `loggerConfig` correct here?

File: packages/graphql-server/src/functions/graphql.ts [hunk 3]
```diff
@@ -59,71 +59,120 @@ export type RedwoodGraphQLContext = {
 /**
  * Options for request and response information to include in the log statements
  * output by UseRedwoodLogger around the execution event
+ *
+ * @param data - Include response data sent to client.
+ * @param operationName - Include operation name.
+ * @param requestId - Include the event's requestId, or if none, generate a uuid as an identifier.
+ * @param query - Include the query. This is the query or mutation (with fields) made in the request.
+ * @param tracing - Include the tracing and timing information.
+ * @param userAgent -Include the browser (or client's) user agent.
```
Comment (thedavidprice) [inline packages/graphql-server/src/functions/graphql.ts:68]:
```suggestion
 * @param userAgent - Include the browser (or client's) user agent.
```

**Review by thedavidprice — APPROVED** [review]
@dthyresson that's a long var name, but descriptive! Good to go for now and we can adjust as needed if/when this goes forward.

A few minor comments and questions in the code. Merge when you're ready.

**Review by dac09 — APPROVED** [review]
LGTM. Explains the billion-line log at start too!

**Comment by dthyresson** [comment]
@dac09 Thanks! I am adding one small defensive update to the api/apollo logger -- just in case a logger isn't setup. Running local e2e now and will update.

**Comment by thedavidprice** [comment]
@dthyresson that E2E error was something broken with Cypress. I just re-triggered.

**Pull request merged**